controllers/common.go

```diff
@@ -291,8 +296,13 @@ func updateOperatorContextStatus(
 		log.Info("Adding link to apiContext", "key", ctxRef.String())
 
 		var operatorContext v1alpha1.OperatorContext
+		namespace := ""
+
+		if ctxRef.Namespace != nil {
+			namespace = *ctxRef.Namespace
+		}
 
-		key := types.NamespacedName{Name: ctxRef.Name, Namespace: ctxRef.Namespace}
+		key := types.NamespacedName{Name: ctxRef.Name, Namespace: namespace}
```

Comment:
Is providing empty namespace safe for k8s client?

Reply:
Doesn't it use `default` namespace when not set?

Reply:
Set default value to `default` instead of empty

Reply:
If not set, I would not explicitly set to default. I would inherit the namespace from the context.

Reply:
Let's fix this in another PR